### PR TITLE
refactor: migrate globals to managed pointers in preparation for deglobalization 

### PR DESCRIPTION
**Pull request by kittywhiskers**
## Additional Notes

In preparation of moving some Dash-specific globals to `NodeContext` (which cannot contain anything that isn't a pointer), many globals have been transformed into `unique_ptr`s (with the sole exception of `CBLSWorker`, which is a `shared_ptr`). 

To reduce the amount of redundant reference passing, some classes were modified to make a `CConnman` object a member of the class, dereferenced from `NodeContext` during initialisation. 

This could not be done for `CSporkManager` and `CGovernanceManager` as they both rely on Dash's flat database implementation which does not take objects that accept initialisation parameters kindly due to an implicit copy being triggered when dumping database components, so they need their database and management components split out before they can take any arguments.

### CODE REVIEW

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
pls see 733b7b3c4fadda29488fa35c10cd75b296c8369a and 7720157e43fc7705bfe8cd340f6f70401ca8edc0 (and maybe a0c7d347f8e0691d7c67019297e0da63a56b81f9)

**Comment by UdjinM6** [comment]
1. pls see https://github.com/UdjinM6/dash/commit/f6e965cea393d4c9a406544359bcc354d923cd55
2. I'm not sure how to fix 6a89c3a (reverting/dropping it helps)

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]
agree to drop https://github.com/dashpay/dash/commit/6a89c3adeba2b4303e6e8577bd5916b3a6d13976

**Comment by UdjinM6** [comment]
Hmm I think you reverted/dropped changes from my earlier suggestions 733b7b3c4fadda29488fa35c10cd75b296c8369a in the second to last force-push https://github.com/dashpay/dash/compare/81a98af39f7ef71cdcc756548a14158a5d09464c..aa43b9d09d79a789acab317be177df134f9830da so the build is broken again 🤷‍♂️

**Comment by PastaPastaPasta** [comment]
linter fails
```
src/masternode/sync.h:54:5: warning: Class 'CMasternodeSync' has a constructor with 1 argument that is not explicit. [noExplicitConstructor]
[82](https://gitlab.com/dashpay/dash/-/jobs/2843071841#L82)Advice not applicable in this specific case? Add an exception by updating
[83](https://gitlab.com/dashpay/dash/-/jobs/2843071841#L83)IGNORED_WARNINGS in test/lint/lint-cppcheck-dash.sh
[84](https://gitlab.com/dashpay/dash/-/jobs/2843071841#L84)
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
pls see https://github.com/UdjinM6/dash/commit/7bd22fbe07bdad1f327445285b083c1e83a5e80c

File: src/interfaces/node.cpp [hunk 1]
```diff
@@ -57,16 +57,18 @@ class EVOImpl : public EVO
...
 class GOVImpl : public GOV
 {
 public:
-    std::vector<CGovernanceObject> getAllNewerThan(int64_t nMoreThanTime) override
+    void getAllNewerThan(std::vector<CGovernanceObject> &objs, int64_t nMoreThanTime) override
     {
-        return governance.GetAllNewerThan(nMoreThanTime);
+        if (governance == nullptr) {
+            governance->GetAllNewerThan(objs, nMoreThanTime);
+        }
```
Comment (UdjinM6) [inline src/interfaces/node.cpp:71]:
```suggestion
        if (governance == nullptr) return;
        governance->GetAllNewerThan(objs, nMoreThanTime);
```
🙈

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — APPROVED** [review]
re-ACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK

**Pull request merged**